src/renderer.cpp

```diff
@@ -596,25 +625,173 @@ void Renderer::shutdown()
 
     bgfx::shutdown();
 }
+void Renderer::clearShadowMap()
+{
+    // Save current view id
+    int tmpBackendViewId = backendViewId;

```

Comment:
Is this code really needed?

Reply:
I did this because the SetClearFlags API uses the `backendViewId` and I need to have the viewId 0  reserved for rendering the shadowmap (more on why this is needed in the comment below).

Anyway, I removed the need to temporally store the `backendViewId` variable by just calling bgfx::setViewClear directly.

Reply:
Thanks, this logic won't work on FF8 though as in some scenes the view 0 may be inheriting the previous scene. Please consider using another approach. Also, why not clearing the shadow map everytime we start a new frame? Wouldn't that work?

Reply:
I may be missing something but I don't understand how FF8 would use view Id 0 if the `backendViewId` variable now starts at view Id 1 and it is reset to 1 every frame.